warehouse/models/mart/transit_database/dim_mobility_mart_providers.sql

```diff
@@ -0,0 +1,163 @@
+-- Services (i.e. dim_services). Each Service will have its own funding sources,
+-- GTFS feeds, and target counties.
+
+{% set innerDelimiter = "';'" %}
+
+WITH organizations AS (
+   SELECT * FROM {{ ref('dim_organizations') }}
+   WHERE _is_current
+),
+
+services AS (
+    SELECT * FROM {{ ref('dim_services') }}
+    WHERE _is_current
+),
+
+county_geogs AS (
+    SELECT * FROM {{ ref('dim_county_geography') }}
+    WHERE _is_current
+),
+
+orgs_x_hq AS (
+    SELECT * FROM {{ ref('bridge_organizations_x_headquarters_county_geography') }}
+    WHERE _is_current
+),
+
+services_x_funding AS (
+    SELECT * FROM {{ ref('bridge_services_x_funding_programs') }}
+    WHERE _is_current
+),
+
+gtfs_schedules AS (
+    SELECT
+        key,
+        {{ from_url_safe_base64('base64_url') }} AS base64_url
+    FROM {{ ref('dim_gtfs_datasets') }}
+    WHERE _is_current
+        AND type = 'schedule'
+),
+
+-- A faux bridge table that conveniently has organizations mapped to services
+-- mapped to GTFS data
+orgs_x_services_x_gtfs AS (
+    SELECT * FROM {{ ref('dim_provider_gtfs_data') }}
+    WHERE _is_current
+),
+
+-- This query will collect every funding source an Organization has via the
+-- services it provides.
+funding_by_org AS (
+    SELECT
+        orgs.organization_key AS org_key,
+        ARRAY_AGG(services_x_funding.funding_program_name) AS funding_sources
+    FROM
+        orgs_x_services_x_gtfs orgs
+    INNER JOIN
+        services_x_funding ON orgs.service_key = services_x_funding.service_key
+    GROUP BY
+        orgs.organization_key
+),
+
+-- This table has historical data for every year (looks like dating back to
+-- 2018), so let's narrow down that table to *just* the most recent year so
+-- we don't get duplicate rows for each year.
+--
+-- We cannot use `_is_current` here because every year is marked as "current"
+-- since it's the "current" record for the respective year.
+annual_ntd AS (
+    SELECT * FROM {{ ref('dim_annual_ntd_agency_information') }}
+    WHERE state = "CA"
+
+    -- We only want data from the latest data from NTD. In the rare edge case
+    -- that an NTD ID is removed and no longer exists in the current year's
+    -- dataset, then replace `DENSE_RANK()` with `ROW_NUMBER()` and add a
+    -- `PARTITION BY ntd_id` inside of the `OVER()`.
+    QUALIFY DENSE_RANK() OVER(ORDER BY year DESC) = 1
+),
+
+gtfs_urls_by_org AS (
+    SELECT
+        orgs_x_services_x_gtfs.organization_key AS org_key,
+        ARRAY_AGG(gtfs_schedules.base64_url) AS gtfs_uris
+    FROM orgs_x_services_x_gtfs
+    LEFT JOIN
+        gtfs_schedules ON orgs_x_services_x_gtfs.schedule_gtfs_dataset_key = gtfs_schedules.key
+    WHERE
+        gtfs_schedules.base64_url IS NOT NULL
+    GROUP BY
+        orgs_x_services_x_gtfs.organization_key
+),
+
+serviced_counties_by_org AS (
+    SELECT
+        orgs.organization_key AS org_key,
+        ARRAY_CONCAT_AGG(services.operating_counties) AS operating_counties
```

Comment:
This one is the most ambiguous re: whether it's worth doing the `distinct` here or later